torch/_guards.py

```diff
@@ -456,24 +456,36 @@ def detect_fake_mode(inputs: Any = None):
     from torch._subclasses.fake_tensor import FakeTensor, FakeTensorMode
     from torch.utils._pytree import tree_flatten
 
+    fake_modes = []
+
     context = TracingContext.get()
     if context is not None:
         fake_mode = context.fake_mode
         if fake_mode is not None:
-            return fake_mode
+            fake_modes.append((fake_mode, lambda: "tracing context"))
```

Comment:
nit: the creation of all the lambdas means we're doing extra work even when we do not hit error case. could fetch the contexts only if we discover multiple fake modes